### PR TITLE
CI: Add `finch-tensor` tests job

### PR DESCRIPTION
**Pull request by mtsokol**
Addresses: https://github.com/willow-ahrens/Finch.jl/issues/539

Hi @willow-ahrens @hameerabbasi,

This PR adds CI job that runs downstream `finch-tensor` tests against new changes in Finch.jl to catch regressions.

### CODE REVIEW

**Comment by mtsokol** [comment]
The PR is ready from my side. 

Another PR (not for merging) shows how it works for catching regressions: https://github.com/willow-ahrens/Finch.jl/pull/542 contains one more commit that renames `SparseCOO` to `SuperSparseCOO`. `CI / Python tests` fails as `finch-tensor` uses previous name.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -89,3 +89,36 @@ jobs:
...
+          ref: 'main'
+          path: 'finch-tensor'
+      - name: Install package
+        run: |
+          cd finch-tensor
+          poetry install --with test
+      - name: Run finch-tensor tests
+        run: |
+          cd finch-tensor
+          FINCH_REPO_PATH=${GITHUB_WORKSPACE} poetry run pytest tests/
```
Comment (hameerabbasi) [inline .github/workflows/CI.yml:124]:
```suggestion
          FINCH_REPO_PATH="${GITHUB_WORKSPACE}" poetry run pytest tests/
```

Spaces in `${GITHUB_WORKSPACE}` may break this without the quotes.

**Review by hameerabbasi — APPROVED** [review]

**Comment by willow-ahrens** [comment]
Let's add instructions in Contributing.md for how to run python tests locally

**Comment by mtsokol** [comment]
I've added a docs section about Python test suite.

**Review by willow-ahrens — COMMENTED** [review]
looks good to me! I'll add this to the queue to be merged

**Pull request merged**